src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const dateParts = date.split(separator);
+
+  const dateObject = {};
+
+  fromFormat.forEach((part, index) => {
+    dateObject[part] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateObject['YY'] = dateObject['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateObject['YY'], 10);
+
+    dateObject['YYYY'] =
+      year < 30 ? '20' + dateObject['YY'] : '19' + dateObject['YY'];
+  }
+
+  const newSeparator = toFormat[3];
```

Comment:
Similar to the 'fromFormat', the code assumes the separator is at index 3 of the 'toFormat' array. This could lead to incorrect behavior if the format changes. Consider handling the separator more dynamically.